pkg/packmgr/installer.go

```
@@ -90,20 +98,66 @@ func Install(ctx context.Context, root, sourcePath string) (InstalledRecord, err
+// 的备份目录：任一步失败都把备份恢复回原位，成功后才删除备份。
+// （Windows 不支持 rename 覆盖已存在目录，因此走"移走旧的→移入新的"两步。）
+func publishStage(ctx context.Context, root, targetDir, stageDir string) (InstalledRecord, error) {
+	backupDir, err := reserveBackupDir(root, targetDir)
+	if err != nil {
 		return InstalledRecord{}, err
 	}
+	restore := func() {
+		if backupDir == "" {
+			return
+		}
+		os.RemoveAll(targetDir)
```

golangci-lint (2.12.2)

[error] Error return value of `os.RemoveAll` is not checked

(errcheck)